src/core/Core.js

```diff
@@ -536,12 +584,30 @@ class Uppy {
 
     this.emit('core:upload')
 
+    const waitingFileIDs = Object.keys(this.state.files)
+      .map(getFile, this)
+      .filter(isFileWaiting)
+      .map(toFileID)
+    function getFile (fileID) {
+      return this.state.files[fileID]
+    }
+    function isFileWaiting (file) {
+      // TODO: replace files[file].isRemote with some logic
+      //
+      // filter files that are now yet being uploaded / haven’t been uploaded
+      // and remote too
+      return !file.progress.uploadStarted || file.isRemote
+    }
+    function toFileID (file) {
+      return file.id
+    }
```

Comment:
This bunch of code makes sure that each plugin in the pipeline works on the same files. I'm not sure if this is the way to go, perhaps the previous way where plugins pick the files they need to work on is better. With this patch there is some implicit state in this function closure…